src/classes/frModel.cls

```diff
@@ -56,32 +72,84 @@ public with sharing abstract class frModel {
+                //if null, we can sync the incoming data.  Otherwise, we don't overwrite
+                fieldsToQueryFor.add(mapping.sf_Name__c);
             }
+            if(!needLastModifiedDate && mapping.Conflict_Resolution__c == MAPPING_OVERWRITE_RECENT) {
+                needLastModifiedDate = true;
+            }
+        }        
+        if(needLastModifiedDate) {
+        	fieldsToQueryFor.add('LastModifiedDate');    
         }
-
-        for(String fieldName : frNameToSfName.keySet()) {
+        
+        String funraiseId = getFunraiseId();
+        String salesforceId = getSalesforceId();
+        SObject queriedFieldsRecord = null;
+        if(!fieldsToQueryFor.isEmpty()) {
+            String query = 'SELECT ' + String.join(new List<String>(fieldsToQueryFor), ',') 
+                + ' FROM ' + sObjectName + ' WHERE ';
+            if(getSalesforceId() != null) {
+                query += 'Id = :salesforceId';
+            } else {
+                query += 'funraise__fr_Id__c = :funraiseId';
+            }
+            List<SObject> results = Database.query(query);
+            if(results.size() > 0) {
+                queriedFieldsRecord = results.get(0);
+            }    
+        }
+        
+        Boolean isMoreRecent = false;
+        if(needLastModifiedDate && queriedFieldsRecord != null) {
+            Datetime sfRecordUpdtime = (Datetime)queriedFieldsRecord.get('LastModifiedDate');
+            isMoreRecent = getUpdtime() != null ? getUpdtime() > sfRecordUpdtime : false;
+        }
+        
+        for(String fieldName : frFieldToMappings.keySet()) {
             if(request.containsKey(fieldName)) {
-                for(frMapping__c mapping : frNameToSfName.get(fieldName)) {
+                for(frMapping__c mapping : frFieldToMappings.get(fieldName)) {
                     Schema.SObjectField field = fields.get(mapping.sf_Name__c);
-                    write(record, field, mapping.sf_Name__c, request.get(fieldName), funraiseId);                    
+                    Object incomingValue = request.get(fieldName);
+                    Boolean write = false;
+                    if(String.isBlank(mapping.Conflict_Resolution__c) || mapping.Conflict_Resolution__c == MAPPING_OVERWRITE) {
+                    	write = true;	    
+                    } else if (mapping.Conflict_Resolution__c == MAPPING_OVERWRITE_NON_NULL && (incomingValue != null || (incomingValue instanceOf String && String.isNotBlank((String)incomingValue)))) {
+                        write = true;
+                    } else if (mapping.Conflict_Resolution__c == MAPPING_NO_OVERWRITE) {
+                        Object existingValue = queriedFieldsRecord != null ? queriedFieldsRecord.get(field) : null;
+                        write = existingValue == null;
+                    } else if (mapping.Conflict_Resolution__c == MAPPING_OVERWRITE_RECENT && isMoreRecent) {
+                        write = true;
+                    }
```

Comment:
Should we add logging/exceptions/something for if the conflict resolution value is none of those?

Reply:
Ha, this was actually a cause of a unit test failure that took me a while to unwind.  I had specified the default value for the Conflict_Resolution__c field and it didn't match the defined values in code and never hit the correct branch in here.  I think adding a log is a good idea